Code/Editor/Core/QtEditorApplication.cpp

```diff
@@ -456,55 +437,16 @@ namespace Editor
 
     void EditorQtApplication::EnableOnIdle(bool enable)
     {
+        m_applicationActive = enable;
         if (enable)
         {
-            if (m_idleTimer->interval() == UninitializedFrequency)
-            {
-                ResetIdleTimerInterval();
-            }
-
-            m_idleTimer->start();
-        }
-        else
-        {
-            m_idleTimer->stop();
+            QTimer::singleShot(0, this, &EditorQtApplication::maybeProcessIdle);
```

Comment:
If `EnableOnIdle(true)` is called twice within the same frame, it would call `maybeProcessIdle` twice.

Shouldn't we protect against queuing the maybeProcessIdle event based on if  `m_applicationActive` transitioning from false to true

Reply:
I suppose so, but this seems like a big edge case.

The only places we call `EnableOnIdle()` are:
1. From `CryEditMain()`, right before calling `QApplication::exec()`. It always enables the idle processing.
2. From the Editor's `MainWindow::closeEvent()`, called with `false` to disable idle processing while the window is closing.
3. From `PyIdleEnable()`, exposed to the behavior context as `idle_enable`. This appears to be called from a ton of independent tests, but always as the first thing they do, and always with `True`.

It seems like anything that is within the idle processing itself won't care about enabling or disabling the idle processing, so that should be managed at a higher level anyway. Would you agree?

Reply:
That BehaviorContext exposure is what is dangerous.
Newer scripts could invoke `EnableOnIdle(true)` twice .

Reply:
Scripts could do that, but are they going to do that from code that executes within `maybeProcessIdle()`? In other words, are those scripts going to be running as part of the Application tick? If they are, it seems they would have no need to enable or disable idle processing of the Editor. If they aren't, then there won't be duplicate calls to `maybeProcesIdle()` as a result of multiple calls to `EnableOnIdle(true)`. What will happen is that 2 QTimerEvents to call the same method will end up on the Qt event loop, and Qt compresses those events into only one event. Either way, it seems like the only danger is that we get 1 frame where the UI lags for 1 frame. I definitely wouldn't expect anything to be manipulating the application tick enabled state in the Editor on a per-frame basis.

Reply:
Is Qt compressing those events based on the function detecting that the function pointer of `&EditorQtApplication::maybeProcessIdle` is the same.